src/app/auth/register/verify/validate/same-as-email.directive.ts

```diff
@@ -0,0 +1,14 @@
+import { AbstractControl, ValidatorFn, ValidationErrors, Validator, NG_VALIDATORS } from "@angular/forms";
+
+/**
+ * Validates that email and display name are not equivalent
+ * @param email User email handle
+ * @returns Valid -> null
+ *          Invalid -> InvalidDisplayName Error
+ */
+export function notEmailValidator(email: string): ValidatorFn {
```

Comment:
public static matchValues(
    matchTo: string // name of the control to match to
  ): (AbstractControl) => ValidationErrors | null {
    return (control: AbstractControl): ValidationErrors | null => {
      return !!control.parent &&
        !!control.parent.value &&
        control.value === control.parent.controls[matchTo].value
        ? null
        : { isMatching: false };
    };
}

Reply:
https://stackoverflow.com/a/55929868

Reply:
This all also doesn't need to be in a separate file since we're not reusing the logic